examples/advanced/examples/foundry_fork_db.rs

```diff
@@ -0,0 +1,158 @@
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let anvil = Anvil::new().spawn();
+    let provider = ProviderBuilder::new().network::<AnyNetwork>().on_http(anvil.endpoint_url());
+
+    let block =
+        provider.get_block(BlockId::latest(), BlockTransactionsKind::Hashes).await?.unwrap();
+
+    // The `BlockchainDbMeta` is used a identifier when the db is flushed to the disk.
+    // This aids in cases where the disk contains data from multiple forks.
+    let meta = BlockchainDbMeta::default()
+        .with_chain_id(31337)
+        .with_block(&block.inner)
+        .with_url(&anvil.endpoint());
+
+    let db = BlockchainDb::new(meta, None);
+
+    // Spawn the backend with the db instance.
+    // `SharedBackend` is used to send request to the `BackendHandler` which is responsible for
+    // filling missing data in the db, and also deduplicate requests that are being sent to the
+    // RPC provider.
+    //
+    // For example, if we send two requests to get_full_block(0) simultaneously, the
+    // `BackendHandler` is smart enough to only send one request to the RPC provider, and queue the
+    // other request until the response is received.
+    // Once the response from RPC provider is received it relays the response to both the requests
+    // over their respective channels.
+    //
+    // The `SharedBackend` and `BackendHandler` communicate over an unbounded channel.
+    let shared = SharedBackend::spawn_backend(Arc::new(provider.clone()), db, None).await;
+
+    let start_t = std::time::Instant::now();
+    let block_rpc = shared.get_full_block(0).unwrap();
+    let time_rpc = start_t.elapsed();
+
+    // `SharedBackend` is cloneable and holds the channel to the same `BackendHandler`.
+    #[allow(clippy::redundant_clone)]
+    let cloned_backend = shared.clone();
+
+    // Block gets cached in the db
+    let start_t = std::time::Instant::now();
+    let block_cache = cloned_backend.get_full_block(0).unwrap();
+    let time_cache = start_t.elapsed();
+
+    assert_eq!(block_rpc, block_cache);
+
+    println!("-------get_full_block--------");
+    // The backend handle falls back to the RPC provider if the block is not in the cache.
+    println!("1st request     (via rpc): {:?}", time_rpc);
+    // The block is cached due to the previous request and can be fetched from db.
+    println!("2nd request (via fork db): {:?}\n", time_cache);
+
+    let alice = anvil.addresses()[0];
+    let bob = anvil.addresses()[1];
+
+    let basefee = block.header.base_fee_per_gas.unwrap();
+
+    let tx_req = TransactionRequest::default()
+        .with_from(alice)
+        .with_to(bob)
+        .with_value(U256::from(100))
+        .with_max_fee_per_gas(basefee as u128)
+        .with_max_priority_fee_per_gas(basefee as u128 + 1)
+        .with_gas_limit(21000)
+        .with_nonce(0);
+
+    let mut evm = configure_evm_env(block, shared.clone(), configure_tx_env(tx_req));
```

Comment:
Let's say this fork environment is configured / updated on every block update, how would that look like? I think this is a common scenario.